### PR TITLE
Set token expiration time when credentials provider is AWSMobileClient

### PR DESCRIPTION
**Pull request by sirknightj**
*Issue #, if available:*

*Description of changes:*
Updated the updateCredentials() method to take AWSMobileClient credentialsProvider into account. When the credentialsProvider is an instanceof AWSMobileClient, the returned expiration time is indefinite, meaning that this method does not get called in the future when the credentials expire and needs refreshing.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by disa6302 — APPROVED** [review]
How many hours of run did we test with this change?

**Comment by sirknightj** [comment]
> How many hours of run did we test with this change?

I left this running overnight, so about 8. Previously, it would stop after ~80 minutes, so we know that this is refreshing regularly.

**Comment by disa6302** [comment]
Perfect. Thanks for confirming

File: aws-android-sdk-kinesisvideo/src/main/java/com/amazonaws/mobileconnectors/kinesisvideo/auth/KinesisVideoCredentialsProviderImpl.java [hunk 1]
```diff
@@ -67,6 +69,15 @@ protected KinesisVideoCredentials updateCredentials() throws KinesisVideoExcepti
 
             expiration = cognitoCredentialsProvider.getSessionCredentialsExpiration();
             log.debug("Refreshed token expiration is %s", expiration);
+        } else if (credentialsProvider instanceof AWSMobileClient) {
+            AWSMobileClient awsMobileClient = (AWSMobileClient) credentialsProvider;
+            try {
```
Comment (poojamat) [inline aws-android-sdk-kinesisvideo/src/main/java/com/amazonaws/mobileconnectors/kinesisvideo/auth/KinesisVideoCredentialsProviderImpl.java:74]: What happens to expiration time when credentials provider is none of these checked types?

Comment (sirknightj) [reply aws-android-sdk-kinesisvideo/src/main/java/com/amazonaws/mobileconnectors/kinesisvideo/auth/KinesisVideoCredentialsProviderImpl.java:74]: The expiration date is then left at what it was previously: `KinesisVideoCredentials.CREDENTIALS_NEVER_EXPIRE` (`new Date(Long.MAX_VALUE)`), so this `updateCredentials()` method will basically never execute a second time.

**Review by poojamat — APPROVED** [review]

**Pull request merged**